src/parser.rs

```diff
@@ -658,6 +685,9 @@ impl<'a> Parser<'a> {
             Token::Caret => Some(BinaryOperator::BitwiseXor),
             Token::Ampersand => Some(BinaryOperator::BitwiseAnd),
             Token::Div => Some(BinaryOperator::Divide),
+            Token::ShiftLeft => Some(BinaryOperator::PGBitwiseShiftLeft),
+            Token::ShiftRight => Some(BinaryOperator::PGBitwiseShiftRight),
+            Token::Sharp => Some(BinaryOperator::PGBitwiseXor),
```

Comment:
I think the natural place for the `dialect_of!` checks is here, not in the tokenizer.